pkg/plugins/platform/local.go

```diff
@@ -812,13 +812,26 @@ func (l *LocalApp) Config() error {
 func (l *LocalApp) Down(removeData bool) error {
```

Comment:
I'm a little uncomfortable with how much nesting and complexity we have in this function now. There are a lot of paths through this function, and along with those paths, lots of room for bugs. It's also much harder to maintain than I would like.

Here are a couple competing ideas for how we might go about making this more maintainable:

- We split out the removeData portion into its own method. I'm not entirely convinced that `Down()` should handle that anyway, as it's arguably separate concern from stopping/removing the containers. Doing this would allow us to simplify the conditionals at the top to something more like the following:

 ```
    if !fileutil.FileExists(l.AppRoot()) || missingConfig != nil {
       return Cleanup(l)
    }

    err := dockerutil.ComposeCmd(l.ComposeFiles(), "down", "-v")
    if err != nil {
        util.Warning("Could not stop site with docker-compose. Attempting manual cleanup.")
        return Cleanup(l)    
    }
```

- We could consider just always calling Cleanup() regardless of whether a docker-compose file exists or not. At this point I don't believe we are using any volumes, so removing a site is just a matter of stopping and removing the containers. If we think we can trust the `Cleanup()` method then it would definitely simplify our code to just rely on it.